### PR TITLE
Fix sign out.

### PR DESCRIPTION
**Pull request by adamjspooner**
@keybase/react-hackers, this is ready for review. It broke when I updated the `UpdatePassword` screen.

cc @zapu

### CODE REVIEW

File: shared/settings/logout/index.tsx [hunk 1]
```diff
@@ -15,133 +15,124 @@ export type Props = {
...
-        </Kb.Box2>
-        {!this.props.checkPasswordIsCorrect ? (
-          <Kb.Box2 direction="vertical">
-            <Kb.ButtonBar align="center" direction="row" fullWidth={true}>
-              {!Styles.isMobile && (
-                <Kb.Button onClick={this.props.onCancel} label="Cancel" fullWidth={true} type="Dim" />
-              )}
+    const inputType = this.state.showTyping ? 'text' : 'password'
+    const keyboardType = this.state.showTyping && Styles.isAndroid ? 'visible-password' : 'default'
+    return this.props.hasRandomPW == null ? (
```
Comment (chrisnojima) [inline shared/settings/logout/index.tsx]: 1. don't ever use `==`
1. just check the falseyness of it. `!this.props.hasRandomPW ? `

File: shared/settings/logout/index.tsx [hunk 2]
```diff
@@ -15,133 +15,124 @@ export type Props = {
...
+          ),
+        ]}
+        footer={{
+          content: !this.props.checkPasswordIsCorrect ? (
+            <Kb.ButtonBar align="center" direction="column" fullWidth={true} style={styles.buttonBar}>
               <Kb.WaitingButton
                 fullWidth={true}
                 waitingKey={Constants.checkPasswordWaitingKey}
-                disabled={!!this.props.checkPasswordIsCorrect}
+                disabled={!!this.props.checkPasswordIsCorrect || this.state.password === ''}
```
Comment (chrisnojima) [inline shared/settings/logout/index.tsx]: same here, lets avoid explicit checks against `''` or null or undefined etc

File: shared/common-adapters/rounded-box.tsx [hunk 3]
```diff
@@ -8,13 +8,14 @@ const Kb = {
 
 type Props = {
   children: React.ReactNode
-  side: 'bottom' | 'middle' | 'top'
+  side?: 'bottom' | 'middle' | 'top'
```
Comment (chrisnojima) [inline shared/common-adapters/rounded-box.tsx]: why make this optional

File: shared/settings/logout/index.tsx (hunk 1 above)
Comment (adamjspooner) [reply shared/settings/logout/index.tsx]: This was just copied over. I wasn't sure if the `==` was the original author's intent. I'll fix it.

File: shared/common-adapters/rounded-box.tsx (hunk 3 above)
Comment (adamjspooner) [reply shared/common-adapters/rounded-box.tsx]: You might want a fully rounded box (as I'm using on the “Do you know your password?” input).

Comment (chrisnojima) [reply shared/common-adapters/rounded-box.tsx]: maybe just add 'all'?

**Pull request merged**